### PR TITLE
ensure we don't block if no ethernet available

### PR DESCRIPTION
**Pull request by sumnerboy12**
`Ethernet.maintain()` maintains the DHCP lease on every program loop - if you pull the ethernet cable this will block while waiting (default 60s) before timing out, thereby crippling the firmware.

Because we are using the W5500 we are able to accurately determine the link status. This change checks this status and ignores `Ethernet.maintain()` and `_mqtt.loop()` (since without ethernet we can't have an MQTT connection) if the link is down.

NOTE: If we move to supporting WiFi we would need to change this and move `_mqtt.loop()` out of this if statement.  

This means if you boot up a Rack32 with no ethernet cable, after the initial 60s timeout when trying to get a lease, the device will continue to operate in "FAILOVER" mode. This would allow local onboard processing, e.g. via my KNX shield, to continue monitoring button events and switching lights.

### CODE REVIEW

**Comment by sumnerboy12** [comment]
I have tested booting up without a cable and letting it timeout getting a DHCP lease. Then plugged the cable in, and on the next program loop it successfully obtained a new DHCP lease, and connected to MQTT.

Also tested, pulling the cable after starting up successfully. The link goes down immediately, but the MQTT library takes a few seconds before it realises the connection is lost. 

We could add the ethernet link status check around the MQTT client, but I figured it was better to leave this to the PubSubClient to figure out on its own. It is less than a minute where MQTT messages are sent and effectively lost, when the cable is pulled.

**Comment by moinmoin-sh** [comment]
I saw the behavior with my breadboard as well but had no time to go in detailed analysis.

Do you really want to wait the 60 sec default? I reduced that to10 sec in an earlier PR which I think is appropriate.

In this context an other question raised. When there is no ethernet or mqtt after boot the FW can't be configured via the retained messages which means a failover won't work properly. Would it make sense to copy a serialized JSON string to a file in SPIFFS and read the config from there? This JSON string could be feed in to config process.

**Comment by sumnerboy12** [comment]
Yeah I read somewhere that the DHCP timeout should be a min of 15 secs so perhaps we drop it down to that for the Rack32?

And you raise a good point about config on startup. Perhaps we just need to monitor the config messages and anytime we receive a retained message, we store it in SPIFS and restore on startup? This means if we startup with no network we get the last retained config, and if the network is there we just get the same retained config applied twice.

Good idea! I might raise this as another issue/PR however.

**Comment by sumnerboy12** [comment]
Reduced timeout to 15s - could you review and approve if you are happy?

**Comment by moinmoin-sh** [comment]
This works good for me (tested on my breadboard). Never saw the 15 sec recommendation but I'm ok with it.

That what you described regarding config/local storage is exactly what I had in mind.

For the failover mode : may be we should turn the colors of both RX/TX mqtt leds to red when the mqtt connection is lost. If you like that idea I would update the lcd api with that functionality.

**Review by moinmoin-sh — APPROVED** [review]

**Comment by sumnerboy12** [comment]
Yep that sounds good re. red leds for mqtt.

**Pull request merged**
